pkgs/top-level/all-packages.nix

```diff
@@ -9078,7 +9078,10 @@ in
 
   ocaml-crunch = ocamlPackages.crunch.bin;
 
-  ocamlformat = callPackage ../development/tools/ocaml/ocamlformat { };
+  inherit (callPackage ../development/tools/ocaml/ocamlformat { })
+    ocamlformat # lastest version
+    ocamlformat_0_11_0 ocamlformat_0_12 ocamlformat_0_13_0 ocamlformat_0_14_0
+    ocamlformat_0_14_1 ocamlformat_0_14_2;
```

Comment:
Do we really need all these top-level attributes?

IMHO, using the `override` mechanism would be less ad-hoc.

Reply:
The issue with override is that it is not just about the version number and the hash, but also the dependencies, the use of Dune 1 or 2. So using override would be just more work for everybody (to find again about this).

Reply:
It’s fine to embed all the knowledge in `nixpkgs`. I’m just talking about the interface.

`ocamlformat.override { version = "0.12"; }` vs `ocamlformat_0_12`.

Reply:
OK, thanks for clarifying. There's one last drawback in using the override method which is the reduced discoverability. Given that `ocamlformat` is always supposed to be pinned, the attributes do make sense to me. But I'm fine with any solution.

Reply:
I wonder if `override` would also make the implementation more complicated. The version is not a simple attribute, `src`, `buildInputs` and possibly others need to change with each version.
Also, would that prevent older version from being cached ?

Reply:
Probably not a lot more complex: you would have a switch based on a version name to choose the value of the other parameters.

> Also, would that prevent older version from being cached ?

In the main NixOS binary cache, yes.

Reply:
OCamlformat being a tool, I think the cache is important because build dependencies are huge and output is small.
Are there other conventions to expose several versions of a package ?

Reply:
I would say the convention you choose is the most common one.